packages/runtime/container-runtime/src/runWhileConnectedCoordinator.ts

```diff
@@ -88,14 +88,16 @@ export class RunWhileConnectedCoordinator implements ICancellableSummarizerContr
             }
             this.stop("summarizerClientDisconnected");
         });
+        this.runtime.once("dispose", () => this.stop("summarizerClientDisconnected"));
     }
 
     /**
      * Starts and waits for a promise which resolves when connected.
      * The promise will also resolve if stopped either externally or by disconnect.
      */
-    public async waitStart() {
+    protected async waitStart() {
         if (!this.runtime.connected && !this.everConnected) {
+            assert(!this._cancelled, "not cancelled");
```

Comment:
Nit: could the assert message be more explanatory?

Reply:
Actually, on second thought, I do not think this assert adds anything (it just asserts current structure of the code, but it if changes, this method would still be correct). So I'll remove it.